### PR TITLE
Enable cgroup 8G memory limit on RPM

### PR DESCRIPTION
**Pull request by brilong**

### CODE REVIEW

File: install/joy.service [hunk 1]
```diff
@@ -5,12 +5,12 @@ After=network.target
 
 [Service]
 Environment=CONFIGFILE=/usr/local/etc/joy/options.cfg
-
 ExecStart=/usr/local/bin/joy -x ${CONFIGFILE}
 ExecReload=/bin/kill -HUP $MAINPID
 KillMode=process
 Restart=on-failure
 RestartSec=60s
+MemoryLimit=8G
```
Comment (fliphil) [inline install/joy.service]: Do we have to change this in the future for systems that have > 8GB of ram?

**Comment by brilong** [comment]
As part of my package creation changes, we now install the systemctl joy.service file to /usr/lib/systemd/system/.  If a user needs to modify the service file for local customization, they would copy it to /etc/systemd/system/ and make their modifications.  This is where we could configure joy to use something higher or lower than 8G on specific hosts.

**Comment by brilong** [comment]
Agreed, it’s deprecated upstream, but we’re testing/compiling on RHEL 7 and CentOS 7, so I’m sticking with what’s supported on those platforms.  I think I’m going to remove MemoryLimit=8G in the /usr/lib/systemd file and we can override in with an /etc file on certain hosts.

 

 

On 9/25/17, 1:03 PM, "Philip Perricone" <notifications@github.com> wrote:

 

@fliphil commented on this pull request.

In install/joy.service:
>  ExecStart=/usr/local/bin/joy -x ${CONFIGFILE}
 ExecReload=/bin/kill -HUP $MAINPID
 KillMode=process
 Restart=on-failure
 RestartSec=60s
+MemoryLimit=8G
I see that MemoryLimit is now deprecated. They recommend using MemoryMax instead:
https://www.freedesktop.org/software/systemd/man/systemd.resource-control.html

**Comment by brilong** [comment]
Actually decided to remove the MemoryLimit=8G from the stock joy systemd service file but still enable CPU and Memory accounting for systemctl services.  This allows use of systemd-cgtop to closely follow joy memory and cpu usage.

**Pull request merged**